src/web/wpan-controller/wpan_controller.cpp

```diff
@@ -53,85 +53,111 @@
 {
-    return gScannedNetworkCount;
+    return mScannedNetworkCount;
 }
 
-int WPANController::Start(void)
+const char *WPANController::GetDBusInterfaceName(void)
 {
-    gInterfaceName = new DBusIfname();
-    return gInterfaceName->ProcessReply();
+
+    DBusIfname dbusIfName;
+    int        ret = kWpantundStatus_Ok;
+
+    dbusIfName.SetInterfaceName(mIfName);
+    VerifyOrExit(dbusIfName.ProcessReply() == kWpantundStatus_Ok, ret = kWpantundStatus_InvalidDBusName);
+exit:
+    return ret ? NULL : dbusIfName.GetDBusName();
 }
 
 int WPANController::Leave(void)
 {
     DBusLeave leaveNetwork;
-
-    leaveNetwork.SetDestination(gInterfaceName->GetDBusName());
+    char      path[DBUS_MAXIMUM_NAME_LENGTH + 1];
+
+    leaveNetwork.SetDestination(GetDBusInterfaceName());
+    leaveNetwork.SetInterfaceName(mIfName);
+    snprintf(path, sizeof(path), "%s/%s", WPANTUND_DBUS_PATH,
+             mIfName);
+    leaveNetwork.SetPath(path);
+    leaveNetwork.SetInterface(WPANTUND_DBUS_APIv1_INTERFACE);
     return leaveNetwork.ProcessReply();
 }
 
 int WPANController::Form(const char *aNetworkname, int aChannel)
 {
     DBusForm formNetwork;
     int      ret = 0;
+    char     path[DBUS_MAXIMUM_NAME_LENGTH + 1];
 
     VerifyOrExit(aNetworkname != NULL, ret = kWpantundStatus_InvalidArgument);
     formNetwork.SetNetworkName(aNetworkname);
     VerifyOrExit((aChannel <= 26 && aChannel >= 11), ret = kWpantundStatus_InvalidArgument);
     formNetwork.SetChannelMask(aChannel);
+    formNetwork.SetInterfaceName(mIfName);
     formNetwork.SetNodeType(LEADER_ROLE);
-    formNetwork.SetDestination(gInterfaceName->GetDBusName());
+    snprintf(path, sizeof(path), "%s/%s", WPAN_TUNNEL_DBUS_PATH,
+             mIfName);
+    formNetwork.SetPath(path);
+    formNetwork.SetDestination(GetDBusInterfaceName());
+    formNetwork.SetInterface(WPAN_TUNNEL_DBUS_INTERFACE);
     ret = formNetwork.ProcessReply();
 exit:
     return ret;
 }
 
-int WPANController::Join(char *aNetworkname,
-                         int16_t aChannel,
+int WPANController::Join(const char *aNetworkname,
+                         uint16_t aChannel,
                          uint64_t aExtPanId,
                          uint16_t aPanId)
 {
     int      ret = 0;
     DBusJoin joinNetwork;
+    char     path[DBUS_MAXIMUM_NAME_LENGTH + 1];
 
     VerifyOrExit(aNetworkname != NULL, ret = kWpantundStatus_InvalidArgument);
     joinNetwork.SetNetworkName(aNetworkname);
     joinNetwork.SetNodeType(ROUTER_ROLE);
+    joinNetwork.SetInterfaceName(mIfName);
+    snprintf(path, sizeof(path), "%s/%s", WPAN_TUNNEL_DBUS_PATH,
+             mIfName);
+    joinNetwork.SetPath(path);
     VerifyOrExit((aChannel <= 26 && aChannel >= 11), ret = kWpantundStatus_InvalidArgument);
     joinNetwork.SetChannel(aChannel);
     VerifyOrExit(aExtPanId != 0, ret = kWpantundStatus_InvalidArgument);
     joinNetwork.SetExtPanId(aExtPanId);
     VerifyOrExit(aPanId != 0, ret = kWpantundStatus_InvalidArgument);
```

Comment:
I noticed this is still not corrected. `0xffff` is invalid, while `0x0000` is valid.